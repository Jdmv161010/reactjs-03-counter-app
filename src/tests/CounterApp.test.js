import { shallow } from "enzyme"
import React from 'react'

import CounterApp from "../CounterApp";

describe('Pruebas al componente CounterApp', () => {

    let wrapper = shallow(<CounterApp/>);

    beforeEach( () => {
        wrapper = shallow(<CounterApp/>);
    });

    test('Evaluacion CounterApp component con snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Evaluacion CounterApp component valor por defecto', () => {
        const wrapper = shallow(<CounterApp value={100}/>);
        const textoDefecto = wrapper.find('h2').text().trim();

        expect(textoDefecto).toBe('100');
    });

    test('debe incrementar con el boton +1', () => {
        wrapper.find('button').at(0).simulate('click');
        const CounterText = wrapper.find('h2').text().trim();
        expect(CounterText).toBe('11');
    });

    test('debe disminuir con el boton -1', () => {
        wrapper.find('button').at(2).simulate('click');
        const CounterText = wrapper.find('h2').text().trim();
        expect(CounterText).toBe('9');
    });

    test('prueba boton reset value', () => {
        const wrapper = shallow(<CounterApp value={100}/>);
        wrapper.find('button').at(1).simulate('click');
        const CounterText = wrapper.find('h2').text().trim();
        
        expect(CounterText).toBe('100');
    });
    
});
