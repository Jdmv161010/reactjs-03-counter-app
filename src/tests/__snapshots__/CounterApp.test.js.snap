// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente CounterApp Evaluacion CounterApp component con snapshot 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
